public void testErrorRecovery() throws Exception
{
    String json = "[ 1, 2, 3, 4, 5, 6, { \"key\": \"value\" }, 8, 9, 10, 11 ]";
    ObjectMapper mapper = new ObjectMapper();
    MappingIterator<Integer> it = mapper.reader(Integer.class).readValues(sequenceInput(json));
    while (it.hasNext()) {
        if (it.nextValue() == 7) {
            throw new JsonProcessingException("Failed!") {};
        }
    }
    // Although we have partially gone through document, we can skip over
    // problematic part and continue
    assertTrue(it.hasNextValue());
    assertEquals(8, it.nextValue().intValue());
}
